src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2213714826>#6f5971</a><span> - 07/08/2024, 04:44 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The exception message change is not critical, but the null check addition is important. However, the null check should also be added before accessing 'head.getNextEntry()' to avoid potential NullPointerException.
</div>
<div id="fix">
<b>Fix:</b> Add a null check for 'head' before accessing 'head.getNextEntry()' to ensure the code does not throw a NullPointerException.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32,7 +32,9 @@ private Entry<K, V> evict() {
         throw new RuntimeException("Cache cannot be empty!");
     }
     Entry<K, V> evicted = head;
 -   head = evicted.getNextEntry();
 +   if (head != null) {
 +       head = evicted.getNextEntry();
 +   }
     if (head != null) {
         head.setPreEntry(null);
     }
```
</div>
</details>
</div>
<br>


</div>